depthai_ros_driver/src/depthai_common.cpp

```diff
@@ -516,18 +503,24 @@ const bool DepthAICommon::set_camera_info_manager(
     return false;
 
   const auto idx = std::distance(_topic_names.cbegin(), it);
+
+  /// Check if camera info manager is initialized
   if (_camera_info_managers[idx])
   {
     return _camera_info_managers[idx]->setCameraName(name) &&
       _camera_info_managers[idx]->loadCameraInfo(uri);
   }
   else
   {
+    auto node_handle = _node_interface.get_node_handle();
     #if defined(USE_ROS2)
-    auto lnh = _node_handle->create_sub_node(name).get();
+    const ROSNodeHandle& nh_ptr = node_handle->create_sub_node(name);
```

Comment:
Add this as well to the node_interface?

Reply:
This is slightly tricky, since the `camera_info_manager` constructer api takes in different arg for the ros.
 - ros1: [ros::NodeHandle](https://github.com/ros-perception/image_common/blob/noetic-devel/camera_info_manager/include/camera_info_manager/camera_info_manager.h#L192)
 - ros2: [rclcpp::Node *](https://github.com/ros-perception/image_common/blob/872a6de7b0efffb188e12f9a16ebc9a3b6ecc67e/camera_info_manager/include/camera_info_manager/camera_info_manager.hpp#L186)

If the ros2 camera info manager takes in a shared ptr, we can totally remove the "switch". This is why this exception is being made.

Reply:
Could you add `create_sub_node` in the node_interface then?

Reply:
Sure. added